### PR TITLE
Alerting: notification template group

### PR DESCRIPTION
**Pull request by ppcano**
Currently, the template name in Alerting refers to two different things:

1. The name of the template created in the UI.

    380781193-cb365150-c702-469b-8181-7bcaaa9af6f0

2. The name of the template defined in `{{ define "template-name" }}`. This is the one used then in the contact point UI.

    380782004-6ae4a753-601a-4452-9314-e75e159abf3e

However, these are two distinct but related objects 
 - `1` is a Grafana model to store template information.
 - `2 {{ define "template-name" }}` refers to Alertmanager template notifications.

This PR clarifies the distinction by renaming `1` as notification template groups. 

The changes in this PR are applied in the frontend and docs.

Screenshot 2024-11-14 at 12 16 11

Screenshot 2024-11-14 at 12 16 20

## TODOs
- [ ] Apply the following [OpenAPI changes](https://gist.github.com/ppcano/c96f0d96d62a6586e4eb33c635e66dc0) and run `make swagger-gen`. Unfortunately, I could not get working `swagger` in my local machine.

### CODE REVIEW

File: public/app/features/alerting/unified/components/receivers/TemplateForm.tsx [hunk 1]
```diff
@@ -350,12 +350,22 @@ function TemplatingBasics() {
   const styles = useStyles2(getStyles);
 
   return (
-    <Alert title="How to" severity="info">
+    <Alert title="" severity="info">
       <Stack direction="row">
         <div>
-          Grafana uses Go templating language to create notification messages.
+          Notification templates use the Go templating language to create notification messages.
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/receivers/TemplateForm.tsx:356]:
```suggestion
          Notification templates use Go templating language to create notification messages.
```

File: public/app/features/alerting/unified/components/receivers/TemplateForm.tsx [hunk 2]
```diff
@@ -350,12 +350,22 @@ function TemplatingBasics() {
...
+          In Grafana, a template group can define multiple notification templates using {'{{'} define "{'<NAME>'}"{' '}
+          {'}}'}.
+          <br />
+          These templates can then be used in contact points and within other notification templates by calling {
+            '{{'
+          }{' '}
+          template "{'<NAME>'}" {'}}'}.
+          <br />
+          <br />
+          For detailed information about notification templates, please visit our documentation.
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/receivers/TemplateForm.tsx:368]:
```suggestion
          For detailed information about notification templates, refer to our documentation.
```

File: public/app/features/alerting/unified/components/receivers/form/fields/TemplateSelector.tsx [hunk 3]
```diff
@@ -129,10 +129,10 @@ function TemplateSelector({ onSelect, onClose, option, valueInForm }: TemplateSe
 
   const templateOptions: Array<SelectableValue<TemplateFieldOption>> = [
     {
-      label: 'Select existing template',
-      ariaLabel: 'Select existing template',
+      label: 'Select notification template',
+      ariaLabel: 'Select notification template',
       value: 'Existing',
-      description: `Select a single template and preview it, or copy it to paste it in the custom tab. ${templateOption === 'Existing' ? 'Clicking Save will save your changes to the selected template.' : ''}`,
+      description: `Select an existing notification template and preview it, or copy it to paste it in the custom tab. ${templateOption === 'Existing' ? 'Clicking Save will save your changes to the selected template.' : ''}`,
```
Comment (brendamuir) [inline public/app/features/alerting/unified/components/receivers/form/fields/TemplateSelector.tsx:135]:
```suggestion
      description: `Select an existing notification template and preview it, or copy it to paste it in the custom tab. ${templateOption === 'Existing' ? 'Clicking Save saves your changes to the selected template.' : ''}`,
```

**Review by brendamuir — APPROVED** [review]
lgtm from docs pov

**Comment by gillesdemey** [comment]
Amazing Pepe! 👏 

I've appended some translations but the rest of the PR looks good to me :shipit:

**Review by gillesdemey — APPROVED** [review]

**Pull request merged**